libsql-server/src/auth/user_auth_strategies/mod.rs

```diff
@@ -2,17 +2,26 @@ pub mod disabled;
 pub mod http_basic;
 pub mod jwt;
 
-use axum::http::HeaderValue;
+use anyhow::Context;
 pub use disabled::*;
 pub use http_basic::*;
 pub use jwt::*;
 
 use super::{AuthError, Authenticated};
 
 pub struct UserAuthContext {
-    pub user_credential: Option<HeaderValue>,
+    pub scheme: Option<String>,
+    pub token: Option<String>, // token might not be required in some cases
 }
 
+impl TryFrom<&str> for UserAuthContext {
```

Comment:
Thought, not sure if we should do this or not but we could implement `FromStr` instead of `TryFrom`? It would give us access to https://doc.rust-lang.org/stable/std/primitive.str.html#method.parse which seems like a better way to express what we are doing here?